core/src/odbc_uri.rs

```diff
@@ -45,6 +45,12 @@ lazy_static! {
         .unwrap();
 }
 
+#[derive(Debug)]
+pub struct UserOptions {
+    pub client_options: ClientOptions,
+    pub connection_string: Option<ConnectionString>,
```

Comment:
```suggestion
    uuid_repr: Option<UuidRepresentation>,
```

Otherwise, there are actually no changes compared to how it was before. Except that now the info is in a struct and not a pair.
If we don't expose the ConnectionString at all, then it makes it safer. Plus it means that there will be no problem with info being different between the ConnectionString and ClientOptions. Right now, ClientOptions is the source of truth, ConnectionString is only used to retrieve uuidRep because of some quirk with the Rust driver, this might not be so straightforward for someone looking at this in a few years if no one from the original team remembers and/or is around.

Reply:
Ah, I see. Easy enough change, I only returned the entire `ConnectionString` to mirror the `ClientOptions`. I've made the change!